### PR TITLE
Fix typos in code comments and Kconfig help texts.

### PR DESCRIPTION
**Pull request by hartmannathan**
## Summary

STM32 (various families): Fix typos in code comments and Kconfig help texts.

## Impact

Improves readability. No functional changes.

## Testing

nxstyle.

### CODE REVIEW

File: arch/arm/src/stm32l5/stm32l5_serial.c [hunk 1]
```diff
@@ -1795,9 +1795,9 @@ static int stm32l5serial_interrupt(int irq, FAR void *context, FAR void *arg)
        * "           "    USART_ISR_ORE  Overrun Error Detected
        * USART_CR3_CTSIE  USART_ISR_CTS  CTS flag               (not used)
        *
-       * NOTE: Some of these status bits must be cleared by explicity writing
-       * one to the ICR register: USART_ICR_CTSCF, USART_ICR_LBDCF. Note of
-       * those are currently being used.
+       * NOTE: Some of these status bits must be cleared by explicitly
+       * writing one to the ICR register: USART_ICR_CTSCF, USART_ICR_LBDCF.
+       * Note of those are currently being used.
```
Comment (saramonteiro) [inline arch/arm/src/stm32l5/stm32l5_serial.c:1800]:
```suggestion
       * None of those are currently being used.
```
Shouldn't here be None?

File: arch/arm/src/stm32/Kconfig [hunk 2]
```diff
@@ -8244,7 +8244,7 @@ config STM32_ADC_MAX_SAMPLES
 		responsibility to correctly select this value.
 		Since the interfece to update the sampling time is available
```
Comment (gustavonihei) [inline arch/arm/src/stm32/Kconfig:8245]:
```suggestion
		Since the interface to update the sampling time is available
```

**Comment by hartmannathan** [comment]
@saramonteiro @gustavonihei Good catches! Thanks!

**Review by gustavonihei — APPROVED** [review]
LGTM

**Pull request merged**